autometa/common/external/samtools.py

```diff
@@ -42,49 +41,65 @@ def sort(sam, bam, nproc=mp.cpu_count()):
         </path/to/alignment.sam>
     bam : str
         </path/to/output/alignment.bam>
-    nproc : int, optional
+    cpus : int, optional
         Number of processors to be used. By default uses all the processors of the system
 
     Raises
     ------
     TypeError
-        nproc must be an integer greater than zero
+        cpus must be an integer greater than zero
     FileNotFoundError
         Specified path is incorrect or the file is empty
-    ChildProcessError
-        Samtools did not run successfully, returns the return code from subprocessing call
+    subprocess.CalledProcessError
+        Samtools did not run successfully
     """
 
-    if type(nproc) is not int or nproc <= 0:
-        raise TypeError(f"nproc must be an integer greater than zero! Given: {nproc}")
-    if not os.path.exists(sam) or os.stat(sam).st_size == 0:
+    if type(cpus) is not int or cpus <= 0:
+        raise TypeError(f"cpus must be an integer greater than zero! Given: {cpus}")
+    if not os.path.exists(sam) or not os.path.getsize(sam):
         raise FileNotFoundError(
             f"The specified path: {sam} is either incorrect or the file is empty"
         )
-    samtools_out_dir = os.path.dirname(os.path.abspath(bam))
-    err = os.path.join(samtools_out_dir, "samtools.err")
-    out = os.path.join(samtools_out_dir, "samtools.out")
-    with tempfile.TemporaryDirectory() as tempdir:
-        temp_bam = os.path.join(tempdir, os.path.basename(bam))
-        cmd = (
-            f"samtools view -@{nproc} -bS {sam} | samtools sort -@{nproc} -o {temp_bam}"
+    with tempfile.TemporaryDirectory() as tempdir:  # this will delete the temporary files even if program stops in between
+        sort_fpath = os.path.join(tempdir, os.path.basename(bam))
+        # See https://stackoverflow.com/questions/13332268/how-to-use-subprocess-command-with-pipes
```

Comment:
You can reduce the length of the url here for lookup later:

```suggestion
        # See https://stackoverflow.com/q/13332268/13118765
```

<img width="766" alt="Screen Shot 2020-07-17 at 6 42 24 AM" src="https://user-images.githubusercontent.com/25933122/87782748-b7f00a00-c7f8-11ea-833a-0504606daf49.png">